### PR TITLE
[BPK-434] use bpk-component-icon for sort direction icons

### PR DESCRIPTION
**Pull request by janeklb**
- https://github.com/Skyscanner/backpack/commit/bf00596c8a0c941e8e6af80c90b24357f3f1aadf: add `hasChildrenOfType` util (probably a good candidate for `bpk-react-utils`) to ensure children of `BpkDataTable` are `BpkDataTableColumn`s.
- https://github.com/Skyscanner/backpack/commit/6f019c38fb25d9d9034416daa8965d54ac9b5d35: use `bpk-component-icon` for sort direction icons

### CODE REVIEW

File: packages/bpk-component-datatable/src/hasChildrenOfType.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+const hasChildrenOfType = (type, atLeast = 1) => (props, propType, componentName) => {
+  if (Children.count(props[propType]) < atLeast) {
+    const inflectedNoun = atLeast === 1 ? 'child' : 'children';
+    return Error(
+      `${componentName} requires at least ${atLeast} '${getDisplayName(type)}' ${inflectedNoun}.`,
+    );
+  }
+
+  const children = Children.toArray(props[propType]);
+  for (let i = 0; i < children.length; i += 1) {
```
Comment (shaundon) [inline packages/bpk-component-datatable/src/hasChildrenOfType.js]: You can use `React.Children.forEach` here: https://reactjs.org/docs/react-api.html#reactchildrenforeach

**Comment by janeklb** [comment]
hey @shaundon - luckily the component hasn't been announced yet (was just merged and published last week https://github.com/Skyscanner/backpack/pull/370) so I think it's safe to update the prop types. Also, there are no docs yet :/

**Comment by shaundon** [comment]
@janeklb my bad - I confused this with `bpk-component-table`!

File: packages/bpk-component-datatable/src/hasChildrenOfType.js (hunk 1 above)
Comment (janeklb) [reply packages/bpk-component-datatable/src/hasChildrenOfType.js]: The reason for this approach is to be able to return directly out of the iteration block -- can you do `return React.Children.forEach(...)`?

Comment (shaundon) [reply packages/bpk-component-datatable/src/hasChildrenOfType.js]: Ah yeah, good point. I'm surprised eslint didn't object to that, it usually shouts at me when I use a native `for` loop. But as long as it works for you 😄

**Comment by shaundon** [comment]
Updated my earlier comment to address me being silly before and talking about the wrong component

File: packages/bpk-component-datatable/src/hasChildrenOfType.js (hunk 1 above)
Comment (janeklb) [reply packages/bpk-component-datatable/src/hasChildrenOfType.js]: Yeah i was waiting to hear from it :)

**Pull request merged**